### PR TITLE
added the ability to view a page showing articles under that topic

### PR DESCRIPTION
**Pull request by LangaLee**

### CODE REVIEW

File: src/Article-components/Article.jsx [hunk 1]
```diff
@@ -109,14 +117,26 @@ const Article = () => {
     getArticleById(articleId, setArticle, setIsLoading);
     getComments(articleId, setComments);
   }, []);
-
+  const commentsCopy = JSON.parse(JSON.stringify(comments));
+  const [sortedComments, setSortedComments] = useState([]);
+  useEffect(() => {
+    if (comments !== null) {
+      setSortedComments(
+        commentsCopy.sort((a, b) => {
```
Comment (pippaamy) [inline src/Article-components/Article.jsx:125]: isnt this already happening in the back end?

File: src/Article-components/Articles.jsx [hunk 2]
```diff
@@ -19,13 +21,19 @@ const Articles = () => {
   );
   const numOfPages = Math.ceil(articles.length / articlesPerPage);
 
+  const { setUrl } = useContext(urlContext);
+
+  const topic = useParams().topic;
+  const displayProperty = topic || "Popularity";
+  console.log(topic);
```
Comment (pippaamy) [inline src/Article-components/Articles.jsx:28]: delete console logs

File: src/api-calls/getArticles.jsx [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
+
+    func2 !== null ? func2(false) : null;
+
+    if (request === "trending") {
+      const articlesCopy = JSON.parse(JSON.stringify(articles));
+      articlesCopy.sort((a, b) => b.votes - a.votes);
+      func(articlesCopy.slice(0, 3));
+    } else if (request !== undefined) {
+      const articlesCopy = JSON.parse(JSON.stringify(articles));
+      func(articlesCopy.filter((article) => article.topic === request));
```
Comment (pippaamy) [inline src/api-calls/getArticles.jsx:18]: our back end can get the articles by topic - this functionality shouldnt be done here

File: src/api-calls/getArticles.jsx [hunk 4]
```diff
@@ -1,14 +1,22 @@
 import axios from "axios";
 
-const getAllArticles = async (setArticles, setIsLoading) => {
+const getAllArticles = async (func, func2, request) => {
   try {
-    // https://fun-news.onrender.com
     let query = "https://fun-news.onrender.com/api/articles";
```
Comment (pippaamy) [inline src/api-calls/getArticles.jsx:5]: we can add the topic query to this query

File: src/api-calls/getArticles.jsx [hunk 5]
```diff
@@ -1,14 +1,22 @@
...
     const response = await axios.get(query);
     const { articles } = response.data;
-    setArticles(articles);
-    setIsLoading(false);
+
+    func2 !== null ? func2(false) : null;
+
+    if (request === "trending") {
+      const articlesCopy = JSON.parse(JSON.stringify(articles));
+      articlesCopy.sort((a, b) => b.votes - a.votes);
```
Comment (pippaamy) [inline src/api-calls/getArticles.jsx:14]: your backend should already be sorting

**Review by pippaamy — COMMENTED** [review]
Good job - its working well but I would change some of the functionality. You are replicating a lot of functionality that the back end is doing for us

File: src/Article-components/Article.jsx (hunk 1 above)
Comment (LangaLee) [reply src/Article-components/Article.jsx:125]: this neccessary for when a new comment is posted  as  they wont be seeing the comments from the database but the ones optimistically rendered

**Pull request merged**